lib/Analysis/NoiseAnalysis/BFV/Noise.h

```diff
@@ -0,0 +1,91 @@
+#ifndef INCLUDE_ANALYSIS_NOISEANALYSIS_BFV_NOISE_H_
+#define INCLUDE_ANALYSIS_NOISEANALYSIS_BFV_NOISE_H_
+
+#include <algorithm>
+#include <cassert>
+#include <optional>
+#include <string>
+
+#include "llvm/include/llvm/Support/raw_ostream.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/Diagnostics.h"       // from @llvm-project
+
+namespace mlir {
+namespace heir {
+namespace bfv {
+
+// This class could be shared among all noise models that tracks the noise by a
+// single value. Noise model could have different interpretation of the value.
+// In BFV world, most noise model just use a single value, either as bound or
+// as variance.
```

Comment:
Anything stopping us from merging this with `BGV/Noise.h/cpp` ? Maybe a new `cc_library` in `lib/Analysis/NoiseAnalysis` that holds the common state in the `mlir::heir` namespace?